src/test/java/com/puppycrawl/tools/checkstyle/internal/XpathRegressionTest.java

```diff
@@ -210,55 +213,99 @@ public void validateIntegrationTestClassNames() throws Exception {
+     */
     @Test
-    public void validateInputFiles() throws Exception {
-        try (DirectoryStream<Path> dirs = Files.newDirectoryStream(inputDir)) {
+    public void validateInputDir() throws Exception {
+        try (DirectoryStream<Path> dirs = Files.newDirectoryStream(inputDir, dirFilter())) {
             for (Path dir : dirs) {
-                // input directory must be named in lower case
-                assertWithMessage(dir + " is not a directory")
-                        .that(Files.isDirectory(dir))
-                        .isTrue();
-                final String dirName = dir.toFile().getName();
-                assertWithMessage("Invalid directory name: " + dirName)
-                        .that(ALLOWED_DIRECTORY_AND_CHECKS)
-                        .containsKey(dirName);
-
-                // input directory must be connected to an existing test
-                final String check = ALLOWED_DIRECTORY_AND_CHECKS.get(dirName);
-                final Path javaPath = javaDir.resolve("XpathRegression" + check + "Test.java");
-                assertWithMessage("Input directory '" + dir
-                            + "' is not connected to Java test case: " + javaPath)
-                        .that(Files.exists(javaPath))
-                        .isTrue();
-
-                // input files should be named correctly
-                validateInputDirectory(dir);
+                assertWithMessage("Invalid directory name: " + dir.getFileName())
+                        .that(DIR_AND_TEST)
+                        .containsKey(dir.getFileName().toString());
+                validateInputDir(dir, DIR_AND_TEST.get(dir.getFileName().toString()));
             }
         }
     }
 
-    private static void validateInputDirectory(Path checkDir) throws IOException {
-        final Pattern pattern = Pattern.compile("^InputXpath(.+)\\.java$");
-        final String check = ALLOWED_DIRECTORY_AND_CHECKS.get(checkDir.toFile().getName());
-
-        try (DirectoryStream<Path> inputPaths = Files.newDirectoryStream(checkDir)) {
-            for (Path inputPath : inputPaths) {
-                final String filename = inputPath.toFile().getName();
-                if (filename.endsWith("java")) {
-                    final Matcher matcher = pattern.matcher(filename);
-                    assertWithMessage(
-                              "Invalid input file '" + inputPath
-                              + "', expected pattern:" + pattern)
-                            .that(matcher.matches())
-                            .isTrue();
-
-                    final String remaining = matcher.group(1);
-                    assertWithMessage("Check name '" + check
-                                + "' should be included in input file: " + inputPath)
-                            .that(remaining)
-                            .startsWith(check);
-                }
+    private void validateInputDir(Path dir, String testName) throws IOException {
```

Comment:
please inline all methods back to test method.

we do not need to this over-abstraction.